src/Solver.cpp

```diff
@@ -649,6 +649,8 @@ void Solver::initializeCmfd() {
   _cmfd->setFSRMaterials(_FSR_materials);
   _cmfd->setFSRFluxes(_scalar_flux);
   _cmfd->setPolarQuadrature(_polar_quad);
+  _cmfd->setGeometry(_geometry);
+  _cmfd->generateKNearestStencils();
```

Comment:
perhaps ``Cmfd`` needs a method like ``initialize()`` which calls both ``generateKNearestStencil()`` and ``initializeSurfaceCurrents()`` under-the-hood? This would also allow both of those method to be made private.